### PR TITLE
fix(store): improve createFeatureSelector warning

### LINKED ISSUE #2116: New warning about unprovided feature state wrecks unit test output
We have unit tests where we use the state without mocking. With the new warning about undefined feature state, that is given while it shouldn't in case of the router state, our unit test are now littered with this warning. Also it shows up in the debugger console when we run the app.

Warning:  

> The feature name "router" does not exist in the state, therefore createFeatureSelector cannot access it.  Be sure it is imported in a loaded module using StoreModule.forRoot('router', ...) or StoreModule.forFeature('router', ...).  If the default state is intended to be undefined, as is the case with router state, this development-only warning message can be ignored.

Also see the discussion here: [Issue 1897](https://github.com/ngrx/platform/issues/1897)

image

## Expected behavior:
I think it is bad practice to give warnings that should be ignored in some cases. This leads to warning/error fatigue. Also unit tests that have hundreds or even thousands of warnings quickly lose their usefulness.

So please find a way to not warn when there is nothing to warn about, like in the case of the router state. Or add the ability to suppress the warnings for a user provided key, or even remove this warning completely as it does seem to do more harm than good.

## Versions of NgRx, Angular, Node, affected browser(s) and operating system(s):
NgRx 8.3
Ng 8.2.5

### PR DESCRIPTION
**Pull request by timdeschryver**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #2116 

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: modules/store/src/selector.ts [hunk 1]
```diff
@@ -605,9 +605,9 @@ export function createFeatureSelector(
 ): MemoizedSelector<any, any> {
   return createSelector((state: any) => {
     const featureState = state[featureName];
-    if (isDevMode() && featureState === undefined) {
+    if (isDevMode() && featureName !== 'router' && !(featureName in state)) {
```
Comment (Halt001) [inline modules/store/src/selector.ts]: May I ask what the added benefit is of having `featureName !== 'router'`? The ` !(featureName in state))` part already catches the initial state of undefined for the router.

Comment (timdeschryver) [reply modules/store/src/selector.ts]: It's unneeded.
Good catch, thanks!

**Comment by jtcrowson** [comment]
This is awesome @timdeschryver!  Thank you  🎉

**Comment by timdeschryver** [comment]
It was a team effort @jtcrowson 💪

**Pull request merged**